php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -825,14 +825,24 @@ public TypeTree type(Optional<InternalSyntaxToken> questionMarkToken, TypeNameTr
     return new TypeTreeImpl(questionMarkToken.orNull(), typeName);
   }
 
-  public NamespaceNameTree namespaceName(List<Tuple<InternalSyntaxToken, InternalSyntaxToken>> tuples) {
+  public NamespaceNameTree namespaceName(Tuple<InternalSyntaxToken, InternalSyntaxToken> firstTuple, Optional<List<Tuple<InternalSyntaxToken, InternalSyntaxToken>>> listOptional) {
+    List<Tuple<InternalSyntaxToken, InternalSyntaxToken>> tuples = new ArrayList<>();
+    tuples.add(firstTuple);
+    if (listOptional.isPresent()) {
+      tuples.addAll(listOptional.get());
+    }
     NameIdentifierTree lastPartIfOneTuple = new NameIdentifierTreeImpl(tuples.get(0).second());
     return namespaceName(lastPartIfOneTuple, tuples.get(0).first(), tuples.subList(1, tuples.size()));
   }
```

Comment:
```suggestion
  public NamespaceNameTree namespaceName(Tuple<InternalSyntaxToken, InternalSyntaxToken> firstTuple, Optional<List<Tuple<InternalSyntaxToken, InternalSyntaxToken>>> listOptional) {
    NameIdentifierTree lastPartIfOneTuple = new NameIdentifierTreeImpl(firstTuple.second());
    return namespaceName(lastPartIfOneTuple, firstTuple.first(), listOptional.or(Collections.emptyList()));
  }
```
I'm not sure why we want now the first tuple to be part of the list, as both first/second element are provided as the first/second argument of the `namespaceName` method ?
The full tuples list is never used, as we only provide a sublist to `namespaceName`, ignoring the first element we added. So I would simply never add it and provide the original list directly.